lib/factories/createStrategyFactory.js

```diff
@@ -51,12 +53,19 @@ module.exports = function (BPromise, commandPrompt, logger, messages, errors) {
               return self.getAPIName(apis);
             }
 
+            if (err instanceof errors.EmptyFieldError) {
```

Comment:
use a single if to check for both errors. use err.getMessage() or something like this to log the error.